### PR TITLE
Proper error handling for invalid doctype

### PR DESCRIPTION
**Pull request by bryceAebi**
Formerly, when uploading an invalid filetype, the UI would erroneously show that the file uploaded successfully. Now a warning toast appears (bottom right corner in screenshot) and the UI does not transition to the success state.

Screenshot 2024-11-07 at 9 30 02 AM

### CODE REVIEW

File: src/components/Input/FileInput.tsx [hunk 1]
```diff
@@ -1,39 +1,40 @@
...
   iconOnly = false,
   icon,
 }: FileInputProps) => {
-  const hiddenFileInput = useRef<HTMLInputElement>(null)
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
 
   const onClick = () => {
     if (hiddenFileInput.current) {
-      hiddenFileInput.current.click()
+      hiddenFileInput.current.value = ""; // reset in case user is trying reupload
```
Comment (bryceAebi) [inline src/components/Input/FileInput.tsx:27]: Only substantive change in this file

File: src/components/TasksListItem/TasksListItem.tsx [hunk 2]
```diff
@@ -66,32 +66,36 @@ export const TasksListItem = ({
...
-                    setIsOpen(false)
-                    goToNextPageIfAllComplete(task)
+                  onUpload={async (file: File) => {
+                    try {
+                      await uploadDocumentForTask(task.id, file);
+                      setIsOpen(false);
+                      goToNextPageIfAllComplete(task);
+                    } catch (e) {
+                      // If upload fails... no-op
+                    }
```
Comment (bryceAebi) [inline src/components/TasksListItem/TasksListItem.tsx:96]: Only substantive change in this file

**Comment by AndrewAcomb** [comment]
I think we can keep this one on hold since we are removing the document type whitelist for now.

File: src/components/TasksListItem/TasksListItem.tsx (hunk 2 above)
Comment (imkesin) [reply src/components/TasksListItem/TasksListItem.tsx:96]: (Nit - no need to change)

I'm mildly in favor of calling `console.error("<Descriptive message>", e)` when these things happens.

**Review by imkesin — APPROVED** [review]

**Review by doneel — APPROVED** [review]
Lgtm

**Comment by bryceAebi** [comment]
We no longer throw any errors based on document type so closing this

**Pull request closed without merging**